ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -496,9 +496,13 @@ forkBlockForging ::
   BlockForging m blk ->
   m (Thread m Void)
 forkBlockForging IS{..} blockForging =
-  forkLinkedWatcher registry label $
-    knownSlotWatcher btime $
-      \currentSlot -> withRegistry (\rr -> withEarlyExit_ $ go rr currentSlot)
+  forkLinkedWatcherFinalize
+    registry
+    label
+    ( knownSlotWatcher btime $
+        \currentSlot -> withRegistry (\rr -> withEarlyExit_ $ go rr currentSlot)
+    )
+    (finalize blockForging)
```

Comment:
Did we store this `blockForging` in some registry? Should we instead `release` it from there?